### PR TITLE
LTP: Add LTP_EXTRA_CONF_FLAGS for 32-bit compatability mode

### PR DESCRIPTION
**Pull request by richiejp**
This allows --CCFLAGS=-m32 --LDFLAGS=-m32 to be passed to the configure script
causing the LTP to be compiled to a 32-bit binary on x86_64 (the flag is
different or missing on some other architectures). Obviously it allows any
other option to be passed to the configure script as well.

Also tries to install the 32-bit development packages, many of which are
currently missing. To complicate matters; the configure script will mark a
feature as present if only the 64-bit development package is available. So
compilation will fail with cryptic messages about missing libraries or
functions instead of skipping tests which require the missing 32-bit
libraries.

It appears that in most cases the 32-bit version of a library is installed as
part of the normal development package, but is missing a symlink like
library.so -> library.so.1. So we can work around the missing packages if
absolutely necessary.

https://progress.opensuse.org/issues/25598
https://bugzilla.suse.com/show_bug.cgi?id=1061591
https://bugzilla.suse.com/show_bug.cgi?id=1061598

@pevik @metan-ucw

### CODE REVIEW

**Comment by pevik** [comment]
LGTM.

BTW we're working on the same thing, but with different approach - I'm working on extending travis build and one of the features is 32 bit build. I run into problem with incorrect detection of 32 bit libraries in some of `m4/ltp-*.m4` files (it detects library presence via presence of header file, e.g.
 [m4/ltp-keyutils.m4](https://github.com/linux-test-project/ltp/blob/master/m4/ltp-keyutils.m4).

**Comment by metan-ucw** [comment]
Detecting library via header presence is standard practice which backfires in more complicated setups. That is why I wrote to the issue that the simplest solution would be compile and install 32bit and 64bit LTP separately and I expected, for the 32bit compilation, that only 32bit devel libraries would be installed.

**Comment by richiejp** [comment]
I did think about removing the 64-bit libraries for a 32-bit installation, but it makes the install script more complicated. So I would rather just file bug reports about missing 32-bit dev packages to begin with.

**Comment by metan-ucw** [comment]
Let's add the removal as a workaround for now so that we can run the tests until it's fixed.

Or we may as well change the LTP configure scripts in to try to compile a minimal C main() with the particular library so that this problem is fixed once for all.

File: tests/kernel/install_ltp.pm [hunk 1]
```diff
@@ -63,13 +66,15 @@ sub install_dependencies {
 sub install_from_git {
     my $url = get_var('LTP_GIT_URL') || 'https://github.com/linux-test-project/ltp';
     my $tag = get_var('LTP_RELEASE') || '';
+    my $configure = './configure --with-open-posix-testsuite --with-realtime-testsuite';
+    my $extra_flags = get_var('LTP_EXTRA_CONF_FLAGS') || '';
```
Comment (asmorodskyi) [inline tests/kernel/install_ltp.pm]: JFYI get_var function accept second parameter as default value so same thing could be written like 
`get_var('LTP_EXTRA_CONF_FLAGS',''); `
but it is just matter of taste :) your way is ok too

**Comment by pevik** [comment]
Although I agree, Metan, that your case would be easiest, having possibility to smoothly build 32-bit and 64-bit on the same devel machine is also reasonable goal.

**Comment by pevik** [comment]
> Or we may as well change the LTP configure scripts in to try to compile a minimal C main() with the particular library so that this problem is fixed once for all.

@metan-ucw That's what I'm working on... Or any other reliable working solution.

**Comment by metan-ucw** [comment]
@pevik there are two things to consider, one is that we should make the tests up and running in OpenQA as fast as possible, the second is that we shouldn't keep horrible hacks for longterm. So I'm okay with working around the problem if we cannot produce a proper fix in reasonable timeframe. I  guess that reworking a few configure scripts that are making problems to us could be done quickly enough.

**Comment by richiejp** [comment]
I would prefer an upstream solution :-)

I can also work around the problem by adding a few symlinks; we can resort to doing that after a few days I suppose.

**Comment by richiejp** [comment]
@pevik I am guessing you want to use AC_SEARCH_LIBS: https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Libraries.html

**Comment by pevik** [comment]
@richiejp Thanks. I was using `AC_CHECK_LIB`, as a diff would be shorter, but `AC_SEARCH_LIBS` might allows us to get rid of some of `m4/ltp-*.m4` files.

**Pull request merged**